src/Entity.php

```diff
@@ -585,4 +585,23 @@ protected function getResolvedFields(): array
     {
         return static::fields();
     }
+
+    /**
+     * Make new entity instance with copied fields
+     *
+     * @return Entity
+     */
+    public final function copy(): Entity
+    {
+        $class = get_called_class();
+
+        /** @var Entity $copy */
+        $copy = new $class();
+
+        $records = $this->getValues();
+
+        $copy->values = $records;
```

Comment:
I think we should remove the timestamps here. And, `$records` should be singular (`$record`)